File: reference-implementations/dotnet/TomlSchema/DiagnosticSeverity.cs
namespace TomlSchema;

/// <summary>
/// Severity level for validation diagnostics.
/// </summary>
public enum DiagnosticSeverity
{
    /// <summary>An error that makes validation unsuccessful.</summary>
    Error,
    /// <summary>A non-fatal validation diagnostic.</summary>
    Warning
}

/// <summary>Wire-spelling helpers for <see cref="DiagnosticSeverity"/>.</summary>
public static class DiagnosticSeverityExtensions
{
    /// <summary>Returns the normative wire spelling (<c>error</c> or <c>warning</c>).</summary>
    public static string WireName(this DiagnosticSeverity severity) =>
        severity == DiagnosticSeverity.Error ? "error" : "warning";
}

File: reference-implementations/java/src/main/java/org/tomlschema/DiagnosticSeverity.java
package org.tomlschema;

/**
 * The severity assigned to a validation diagnostic.
 */
public enum DiagnosticSeverity {
    ERROR,
    WARNING;

    /**
     * Returns the normative wire spelling of this severity ({@code error} or
     * {@code warning}).
     *
     * @return the severity name used for conformance comparison
     */
    public String wireName() {
        return name().toLowerCase(java.util.Locale.ROOT);
    }
}
